source/material_model/visco_plastic.cc

```diff
@@ -353,13 +354,24 @@ namespace aspect
     evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
              MaterialModel::MaterialModelOutputs<dim> &out) const
     {
+      // set up additional output for the derivatives
+      MaterialModel::MaterialModelDerivatives<dim> *derivatives;
+      derivatives = out.template get_additional_output<MaterialModel::MaterialModelDerivatives<dim> >();
+
+      // If we use the Newton solver, get the derivative scaling factor,
+      // otherwise set it to zero.
+      double derivative_scaling_factor = 0.0;
+      if (this->get_parameters().nonlinear_solver == Parameters<dim>::NonlinearSolver::Newton_Stokes)
+        derivative_scaling_factor = this->get_newton_handler().get_newton_derivative_scaling_factor();
```

Comment:
Hm, why do you actually need this in the material model? Shouldn't the scaling factor be applied only at the one or two places in the assembly where we compute the Newton matrix? If that were the case, you wouldn't have to deal with it in every single material model.

Reply:
So the factor itself is actually not used to compute something here. It is only used to determine wheter the derivatives should be computed. Because when this scaling factor is zero, which may happen even when `derivatives != NULL`, there is no use in computing the derivatives.